### PR TITLE
Expose metadata field in add service and edit service pages

### PR DESCRIPTION
**Pull request by loganhz**
[Related issue 2243](https://github.com/rancher/rancher/issues/2243)

Before,

image

After focus out,

image

### CODE REVIEW

**Comment by vincent99** [comment]
Loading ace and this is far more involved than I was thinking (and hundreds of KB) for a tiny feature...

**Comment by loganhz** [comment]
@vincent99  agree!
Now it is a `input-multiline`

**Comment by vincent99** [comment]
This looks pretty good but should default to empty, or empty object (`{}`), not `null`.  And the validation should check that the top-level value is an object, an array or literal (`[1,2,3]`, `false`, etc) are not allowed because the field is map of string->json.

File: app/components/json-editor/component.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    } catch (err) {
+      this.set('isInvalid', true)
+    }
+  }.observes('jsonString'),
+
+  parseJSON: function (jsonString) {
+    try {
+      return JSON.parse(jsonString);
+    } catch (err) {
+      return JSON.parse(sanitize(jsonString));
```
[the remark this answers is not in the document]
Comment (loganhz) [reply app/components/json-editor/component.js]: the 2nd parse is expected to throw an error sometimes. the error will be caught in `onChange ` method. Then the `invalid` value will be set to `true` there

**Comment by loganhz** [comment]
updated.
thanks a lot for your review!

**Pull request merged**